File: p9/main2.cpp
#include <iostream>

class A {
    private:
    public:
    // constructor:
    A() {}

    virtual void m1() {
        std::cout << "AAAA" << std::endl;
    }

    void m() {
        this->m1();
    }
};

class B : public A {
    private:
    public:
    B() {}
    
    void m1() {
        std::cout << "BBBB" << std::endl;
    }
};

int main(void) {
    B b;
    b.m(); // muestra AAAA
    return 0;
}

File: p9/Main2.java

class A {
    public A() {}

    public void m1() {
        System.out.println("AAAA");
    }
}

class B extends A {
    public B() {}
    
    public void m1() {
        System.out.println("BBBB");
    }
}

class C extends B {
    public C() {}
    
    public void m1() {
        super.m1();
    }
}

public class Main2 {
    public static void main(String[] args) {
        C c = new C();
        c.m1();
    }
}

// si uso final en un mètodo no lo permite redefinir en una clase que herede (PROBAR)
